### PR TITLE
[12.0][MIG] project_stock: Migration from 14.0 to 12.0

### PR DESCRIPTION
**Pull request by marcelsavegnago**

### CODE REVIEW

File: project_stock/models/project_project.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        help="Default location to which materials are consumed.",
+    )
+    stock_analytic_date = fields.Date(string="Analytic date")
+
+    @api.onchange("picking_type_id")
+    def _onchange_picking_type_id(self):
+        self.location_id = self.picking_type_id.default_location_src_id.id
+        self.location_dest_id = self.picking_type_id.default_location_dest_id.id
+
+    def write(self, vals):
```
Comment (quannguyen1560) [inline project_stock/models/project_project.py:41]: You need @api.multi for version 12

File: project_stock/models/project_task.py [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+                lambda x: x.state not in ("done", "cancel")
+            )._do_unreserve()
+        return True
+
+    def button_unreserve(self):
+        self.ensure_one()
+        self.do_unreserve()
+        return True
+
+    def action_cancel(self):
```
Comment (quannguyen1560) [inline project_stock/models/project_task.py:200]: Should have self.ensure_one()

File: project_stock/models/project_task.py [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+                analytic_line_model.create(move._prepare_analytic_line_from_task())
+
+    def action_see_move_scrap(self):
+        self.ensure_one()
+        action = self.env.ref("stock.action_stock_scrap").read()[0]
+        action["domain"] = [("task_id", "=", self.id)]
+        action["context"] = dict(self._context, default_origin=self.name)
+        return action
+
+    def write(self, vals):
```
Comment (quannguyen1560) [inline project_stock/models/project_task.py:235]: Should have @api.multi

File: project_stock/models/project_task.py [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+            if stage.done_stock_moves:
+                # Avoid permissions error if the user does not have access to stock.
+                self.sudo().action_assign()
+        # Update info
+        field_names = ("location_id", "location_dest_id")
+        if any(vals.get(field) for field in field_names):
+            self._update_moves_info()
+        return res
+
+    def unlink(self):
```
Comment (quannguyen1560) [inline project_stock/models/project_task.py:249]: Should have @api.multi

**Review by quannguyen1560 — CHANGES_REQUESTED** [review]
Please check my comment and fix the pipeline. thank you

**Comment by marcelsavegnago** [comment]
@OCA/community-maintainers Can anyone help or guide me on this case?

https://app.travis-ci.com/github/OCA/project/jobs/595702814#L293

```
ERROR: Could not find a version that satisfies the requirement pylint-plugin-utils==0.7 (from -r /home/travis/maintainer-quality-tools/travis/pylint_odoo_requirements.txt (line 1)) (from versions: 0.1, 0.1.1, 0.2, 0.2.1, 0.2.2, 0.2.3, 0.2.4, 0.2.6, 0.3, 0.4, 0.5, 0.6)
ERROR: No matching distribution found for pylint-plugin-utils==0.7 (from -r /home/travis/maintainer-quality-tools/travis/pylint_odoo_requirements.txt (line 1))
```

I found an issue reporting the same problem in another repository.
https://github.com/OCA/maintainer-quality-tools/issues/712

**Comment by marcelsavegnago** [comment]
ping @OCA/project-service-maintainers

**Comment by marcelsavegnago** [comment]
ping @pedrobaeza @victoralmau

**Comment by marcelsavegnago** [comment]
ping @quannguyen1560

**Comment by quannguyen1560** [comment]
@marcelsavegnago 
Did you add yourself to the CONTRIBUTORS ?

**Comment by marcelsavegnago** [comment]
> @marcelsavegnago Did you add yourself to the CONTRIBUTORS ?

done

**Review by quannguyen1560 — APPROVED** [review]
LGTM

**Pull request merged**